compiler/src/iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -1592,6 +1592,52 @@ getDefaultVscaleRange(IREE::HAL::ExecutableTargetAttr targetAttr) {
   return std::nullopt;
 }
 
+FailureOr<int64_t> getStaticPartOfScalableTileSize(Operation *op) {
+  auto mulIOp = dyn_cast<arith::MulIOp>(op);
+  if (!mulIOp)
+    return failure();
+
+  auto lhs = mulIOp.getLhs().getDefiningOp();
+  auto rhs = mulIOp.getRhs().getDefiningOp();
+
+  auto cstOp = isa<arith::ConstantOp>(lhs) ? cast<arith::ConstantOp>(lhs)
+                                           : dyn_cast<arith::ConstantOp>(rhs);
+  if (!cstOp)
+    return failure();
+  if (!isa<vector::VectorScaleOp>(lhs) && !isa<vector::VectorScaleOp>(rhs))
+    return failure();
+  if (auto integerAttr = dyn_cast<IntegerAttr>(cstOp.getValue()))
+    return integerAttr.getInt();
+  return failure();
+}
+
+FailureOr<SizesAndScalableFlags>
+getScalableTileSizesAndFlags(SmallVector<OpFoldResult> mixedInnerTiles) {
+  SmallVector<int64_t> tileSizes(mixedInnerTiles.size(), 0);
+  IREE::Codegen::ScalableTileFlags scalableFlags(mixedInnerTiles.size(), false);
+  for (unsigned pos = 0; pos < mixedInnerTiles.size(); ++pos) {
+    if (auto innerTileVal = dyn_cast<Value>(mixedInnerTiles[pos])) {
+      FailureOr<int64_t> innerTile =
+          getStaticPartOfScalableTileSize(innerTileVal.getDefiningOp());
+      if (failed(innerTile)) {
+        LDBG() << "Found non-scalable dynamic inner tile!";
+        return failure();
+      }
+      tileSizes[pos] = innerTile.value();
+      scalableFlags[pos] = true;
+      continue;
+    }
+    std::optional<int64_t> innerTile =
+        getConstantIntValue(mixedInnerTiles[pos]);
+    if (!innerTile.has_value()) {
+      LDBG() << "Error while inferring static inner tile size!";
+      return failure();
+    }
```

Comment:
[nit] The two cases were not immediately obvious to me. Perhaps add comments?

```suggestion
    // Scalable tile?
    if (auto innerTileVal = dyn_cast<Value>(mixedInnerTiles[pos])) {
      FailureOr<int64_t> innerTile =
          getStaticPartOfScalableTileSize(innerTileVal.getDefiningOp());
      if (failed(innerTile)) {
        LDBG() << "Found non-scalable dynamic inner tile!";
        return failure();
      }
      tileSizes[pos] = innerTile.value();
      scalableFlags[pos] = true;
      continue;
    }
    // Constant tile?
    std::optional<int64_t> innerTile =
        getConstantIntValue(mixedInnerTiles[pos]);
    if (!innerTile.has_value()) {
      LDBG() << "Error while inferring static inner tile size!";
      return failure();
    }
```